app/controllers/encrypt_controller.rb

```diff
@@ -135,7 +135,8 @@ def update_post
     encrypted_raw = params.require(:encrypted_raw)
 
     post = Post.find_by(id: post_id)
-    guardian.ensure_can_edit!(post)
+
+    raise Discourse::InvalidAccess if post.user != current_user
```

Comment:
If edit isn't the appropriate action do we need a `guardian.ensure_can_encrypt_post!` instead?